src/imu/imu.c

```diff
@@ -71,17 +58,24 @@ void imu_calibrate_zero(){
 		int zeroValue = g_imu_filter[physicalChannel].current_value;
 		//adjust for gravity
 		float countsPerUnit = imu_device_counts_per_unit(physicalChannel);
-		if (c->physicalChannel == IMU_CHANNEL_Z) zeroValue-= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
+		if (i == IMU_CHANNEL_Z) zeroValue -= (countsPerUnit * (c->mode != MODE_IMU_INVERTED ? 1 : -1));
 		c->zeroValue = zeroValue;
 	}
 }
 
 int imu_init(LoggerConfig *loggerConfig){
 	imu_device_init();
-	init_filters(loggerConfig);
+    init_filters(loggerConfig);
 	return 1;
 }
 
+int imu_soft_init(LoggerConfig *loggerConfig){
+    init_filters(loggerConfig);
+    return 1;
+}
```

Comment:
It's a standard convention for HAL / device init methods, even if it always succeeds.